src/rust/catcollar/mod.rs

```diff
@@ -521,7 +521,12 @@ impl CatcollarLibOS {
 
     /// Takes out the operation result descriptor associated with the target scheduler handle.
     fn take_result(&mut self, handle: TaskHandle) -> (QDesc, OperationResult) {
-        let task: OperationTask = OperationTask::from(self.runtime.scheduler.remove(handle).as_any());
+        let task: OperationTask = if let Some(task) = self.runtime.scheduler.remove(&handle) {
+            OperationTask::from(task.as_any())
+        } else {
+            panic!("Removing task that does not exist (either was previously removed or never inserted)");
```

Comment:
Shouldn't this condition be `unreachable!()`? Even better, wouldn't be the case to change the function signature to return a `Fail` and we propagate it back to the PDPIX call?